drivers/led/ncp5623c.c

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2023 NXP Semiconductors
+ * Copyright (c) 2023 Cognipilot Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT onnn_ncp5623c
+
+/**
+ * @file
+ * @brief NCP5623C LED driver
+ *
+ * The NCP5623C is a 3-channel LED driver that communicates over I2C.
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ncp5623c, CONFIG_LED_LOG_LEVEL);
+
+#include "led_context.h"
+
+#define NCP5623_LED_CURRENT 0x20
+#define NCP5623_LED_PWM0    0x40
+#define NCP5623_LED_PWM1    0x60
+#define NCP5623_LED_PWM2    0x80
+
+#define NCP5623C_CHANNEL_COUNT 3
+
+/* Brightness limits in percent */
+#define NCP5623C_MIN_BRIGHTNESS 0
+#define NCP5623C_MAX_BRIGHTNESS 0x1f
+
+static const uint8_t led_channels[] = {NCP5623_LED_PWM0, NCP5623_LED_PWM1, NCP5623_LED_PWM2};
+
+struct ncp5623c_config {
+	struct i2c_dt_spec bus;
+	uint8_t num_leds;
+	const struct led_info *leds_info;
+};
+
+struct ncp5623c_data {
+	struct led_data dev_data;
+};
+
+static const struct led_info *ncp5623c_led_to_info(const struct ncp5623c_config *config,
+						   uint32_t led)
+{
+	int i;
+
+	for (i = 0; i < config->num_leds; i++) {
+		if (config->leds_info[i].index == led) {
+			return &config->leds_info[i];
+		}
+	}
+
+	return NULL;
+}
+
+static int ncp5623c_get_info(const struct device *dev, uint32_t led, const struct led_info **info)
+{
+	const struct ncp5623c_config *config = dev->config;
+	const struct led_info *led_info = ncp5623c_led_to_info(config, led);
+
+	if (!led_info) {
+		return -EINVAL;
+	}
+
+	*info = led_info;
+
+	return 0;
+}
+
+static int ncp5623c_set_color(const struct device *dev, uint32_t led, uint8_t num_colors,
+			      const uint8_t *color)
+{
+	const struct ncp5623c_config *config = dev->config;
+
+	if (led > 0) {
+		return -EINVAL;
+	}
+
+	if (num_colors != NCP5623C_CHANNEL_COUNT) {
+		return -EINVAL;
+	}
+
+	uint8_t buf[6] = {0x70, NCP5623_LED_PWM0, 0x70, NCP5623_LED_PWM1, 0x70, NCP5623_LED_PWM2};
+
+	buf[1] = buf[1] | color[0] / 8;
+	buf[3] = buf[3] | color[1] / 8;
+	buf[5] = buf[5] | color[2] / 8;
+
+	return i2c_burst_write_dt(&config->bus, NCP5623_LED_CURRENT | NCP5623C_MAX_BRIGHTNESS, buf,
+				  sizeof(buf));
+}
+
+static int ncp5623c_led_set_brightness(const struct device *dev, uint32_t led, uint8_t value)
```

Comment:
There is a problem here. the `led` parameter is not consistent between `set_color` and `set_brightness`.

In `set_color`, `led` is a LED number and the only possible value is 0. But here in `set_brightness`, `led` is a channel number and it can take [0-2] as value.

Maybe you could change `set_brightness` to update the "maximum Output LED Current" register instead of the channels PWM ? IIUC this register applies to all the RGB channels and then actually allows to configure the LED brightness.